scripts/actions/fetch-and-deserialize.js

```diff
@@ -53,63 +53,162 @@ const writeFilesSync = (vfiles) => {
+   * @param {String[]} input.fileUris
+   */
+  return async ({ fileUris }) => {
+    const fileUriStr = fileUris.reduce((str, uri) => {
+      return str.concat(`&fileUris[]=${encodeURIComponent(uri)}`);
+    }, '');
+
+    const localeIdStr = `localeIds[]=${locale}`;
+
+    const response = await fetch(
+      `https://api.smartling.com/files-api/v2/projects/${projectId}/files/zip?${localeIdStr}${fileUriStr}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${await getAccessToken()}`,
+        },
+      }
+    );
+
+    if (response.status !== 200) {
+      console.log(
+        `Error encountered downloading files from vendor. Response code received: ${response.status}`
+      );
+      console.log(
+        `The following files were not downloaded for ${locale}: ${JSON.stringify(
+          fileUris,
+          null,
+          4
+        )}`
+      );
+      return null;
+    }
 
-  const zip = new AdmZip(buffer);
-  const zipEntries = zip.getEntries();
+    const buffer = await response.buffer();
+    const zip = new AdmZip(buffer);
 
-  const translatedHtml = zipEntries.map((entry) => {
-    const filepath = entry.entryName.replace(`${locale}/src/content/docs`, '');
-    const slug = filepath.replace(`.mdx`, '');
-    return {
-      path: slug,
-      html: zip.readAsText(entry, 'utf8'),
-    };
-  });
+    return zip;
+  };
+};
 
-  const deserializedMdx = await Promise.all(
-    translatedHtml.map(async ({ path, html }) => {
-      console.log(`[*] Deserializing ${path}`);
+/**
+ * @param {String} locale
+ */
+const extractFiles = (locale) => {
+  /**
+   * @param {AdmZip} zip - the downloaded zip containing batch of files.
+   */
+  return async (zip) => {
+    return zip.getEntries().map((entry) => {
+      const filepath = entry.entryName.replace(
+        `${locale}/src/content/docs`,
+        ''
+      );
+      const slug = filepath.replace(`.mdx`, '');
       return {
-        path: `src/i18n/content/${localesMap[locale]}/docs/${path}`,
-        mdx: await deserializedHtml(html),
+        path: slug,
+        html: zip.readAsText(entry, 'utf8'),
       };
-    })
-  );
+    });
+  };
+};
 
-  const files = deserializedMdx.map(
-    ({ path, mdx }) =>
-      vfile({
+/**
+ * @param {String} locale
+ */
+const deserializeHtmlToMdx = (locale) => {
+  /**
+   * @param {object} file
+   * @param {String} file.path - path of file w/o 'src/.../content/docs' prefix
+   * @param {String} file.html - html content of file
+   */
+  return async ({ path, html }) => {
+    try {
+      const newPath = `src/i18n/content/${localesMap[locale]}/docs/${path}`;
+      const mdx = await deserializedHtml(html);
+
+      const temp = vfile({
         contents: mdx,
-        path,
+        path: newPath,
         extname: '.mdx',
-      }),
-    'utf-8'
+      });
+
+      createDirectories([temp]);
+      writeFilesSync([temp]);
+
+      return { ok: true, slug: `src/content/docs/${path}` };
```

Comment:
will need to fix this path up. 
* missing `.mdx` extension
* will have double `/`